content_scripts/link_hints.coffee

```diff
@@ -121,21 +121,21 @@ HintCoordinator =
     @linkHintsMode = @localHints = null
 
 LinkHints =
-  activateMode: (count = 1, {mode}) ->
-    mode ?= OPEN_IN_CURRENT_TAB
+  activateMode: (count = 1, options) ->
+    mode = options.mode ? OPEN_IN_CURRENT_TAB
     if 0 < count or mode is OPEN_WITH_QUEUE
-      HintCoordinator.prepareToActivateMode mode, (isSuccess) ->
+      HintCoordinator.prepareToActivateMode mode, options, (isSuccess) ->
         if isSuccess
           # Wait for the next tick to allow the previous mode to exit.  It might yet generate a click event,
           # which would cause our new mode to exit immediately.
-          Utils.nextTick -> LinkHints.activateMode count-1, {mode}
+          Utils.nextTick -> LinkHints.activateMode count-1, options, {mode}
 
-  activateModeToOpenInNewTab: (count) -> @activateMode count, mode: OPEN_IN_NEW_BG_TAB
-  activateModeToOpenInNewForegroundTab: (count) -> @activateMode count, mode: OPEN_IN_NEW_FG_TAB
-  activateModeToCopyLinkUrl: (count) -> @activateMode count, mode: COPY_LINK_URL
-  activateModeWithQueue: -> @activateMode 1, mode: OPEN_WITH_QUEUE
-  activateModeToOpenIncognito: (count) -> @activateMode count, mode: OPEN_INCOGNITO
-  activateModeToDownloadLink: (count) -> @activateMode count, mode: DOWNLOAD_LINK_URL
+  activateModeToOpenInNewTab: (count, options) -> @activateMode count, (extend options, mode: OPEN_IN_NEW_BG_TAB)
```

Comment:
Could you check whether the use of `options` here is correct?

I had to tweak something you did recently, see [here](https://github.com/philc/vimium/blob/master/content_scripts/mode_normal.coffee#L64).

Reply:
`options` here is unwrapped into the actual user options in `HintCoordinator.prepareToActivateMode`, to keep these lines simpler. I based that code off your tweak, so I'm pretty sure it's right.